### ISSUE TITLE
rancher2_bootstrap current_password always detects changes

### ISSUE THREAD

**Issue by braunsonm** [rancher/terraform-provider-rancher2#764]
It seems with a simple config:

```hcl
resource "random_password" "bootstrap" {
}

resource "rancher2_bootstrap" "bootstrap" {
  current_password = random_password.bootstrap.result
}
```

Subsequent runs with `terraform apply` always show the `current_password` as being updated in place even though it has not changed.

**Comment by rawmind0** [comment]
This is expected. The `current_password` argument is just to be used if admin password/token has been updated outside terraform. Once rescued, the `current_password` has to be just removed from tf file

**Comment by braunsonm** [comment]
@rawmind0 Not sure I follow. If Rancher is installed using Terraform using this bootstrapped password:

```hcl
resource "helm_release" "rancher_ui" {
  name = "rancher"
  repository = "..."
  chart = "rancher"

  // set the bootstrap password here to random_password.bootstrap.result
}
```

You should not need to modify the .tf file after you've bootstrapped

**Comment by rawmind0** [comment]
@braunsonm you have 2 options: 
* set the Rancher helm bootstrap password to `admin` and then `rancher2_bootstrap` will not require `current_password` argument.
* set `current_password` at `rancher2_bootstrap` to same used at Rancher helm chart, and once you run the bootstrap properly, remove it from tf file to avoid  updated in place at subsequent terraform apply

**Comment by braunsonm** [comment]
That seems like bad design and this should be considered a bug don't you think? 
- You should not be encouraging users to use the default bootstrap password `admin` when provisioning clusters that may be internet facing
- Option 2 doesn't follow Terraforms principals. You should not need to change your `.tf` files based on the state of your infrastructure to prevent noise. The value of `current_password` did not change so it should not be attempting to update it.

**Comment by rawmind0** [comment]
> That seems like bad design and this should be considered a bug don't you think?

No, i don't think it's a bad design nor a bug. It's simply a new feature added to rancher 2.6.0 (not stable rancher version yet) that it's not yet implemented here.  Any contribution would be appreciated 

> - You should not be encouraging users to use the default bootstrap password `admin` when provisioning clusters that may be internet facing

I'm not encouraging users to do nothing, just explaining the available options to bootstrap rancher 2.6.0 with the provider at the moment.

> - Option 2 doesn't follow Terraforms principals. You should not need to change your `.tf` files based on the state of your infrastructure to prevent noise. The value of `current_password` did not change so it should not be attempting to update it.

Right, Terraforms principals are not followed if you use `current_password` in this way. But the use of `current_password` field is just justified as workaround to recover admin password management  when if has been updated outside of terraform, as said in [docs](https://registry.terraform.io/providers/rancher/rancher2/latest/docs/resources/bootstrap#current_password)

**Comment by braunsonm** [comment]
Sorry you stated this is "expected" so that made it sound like you don't consider this as a bug and that you designed it this way? I would personally suggest not viewing the current_password as a change everytime but you mention it was done this way on purpose.

Please feel free to do what you'd like with this issue. Whether that's close it or suggest an implementation change.

**Comment by rawmind0** [comment]
Submitted PR https://github.com/rancher/terraform-provider-rancher2/pull/769 adding the feature

**Comment by rawmind0** [comment]
PR merged. The feature will be available at next terraform provider release, v1.21.0
